locust/render.py

```diff
@@ -188,6 +191,45 @@ def render_change_as_html(
     return E.LI(*change_elements)
 
 
+def render_change_as_compressed_html(
+    change: Dict[str, Any], filepath: str, current_depth: int, max_depth: int
+) -> Optional[Any]:
+    """
+    Returns nested part of report in a compact format.
+    Useful for large changes as it allows to perceive more functions and classes.
+    """
+    if current_depth >= max_depth:
+        return None
+
+    link = change.get("link")
+    if link is None:
+        link = filepath
+
+    change_elements: List[Any] = [
```

Comment:
@kompotkot : I think the best way to reduce duplication is to create a `change_representation_compressed` function and a `change_representation_full` function, which implement the distinct parts of `render_change_as_html`.

Both functions should have signature `(change: Dict[str, Any]) -> Optional[Any]`.

In `render_change_as_html`, you can add an optional enum parameter which chooses between compressed and uncompressed representations.

When we create a file section handler as part of defining our renderers, we should pass the appropriate representation argument down to the appropriate `render_change_*` function.